Fixes metadata on declare in m.m.serialization

john-metabase opened the pull request:
Quick fix for a warning in `metabase.models.serialization`

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/29691)
<!-- Reviewable:end -->

dpsutton commented:
Was:
> Warning: *lookup-by-id* not declared dynamic and thus is not dynamically rebindable, but its name suggests otherwise. Please either indicate ^:dynamic *lookup-by-id* or change the name. (metabase/models/serialization.clj:245)

Now its:
> Warning: *lookup-by-id* not declared dynamic and thus is not dynamically rebindable, but its name suggests otherwise. Please either indicate ^:dynamic *lookup-by-id* or change the name. (metabase/models/serialization.clj:375)

Looks like you need it in both places.

dpsutton reviewed (changes requested):
Need in both places it seems

john-metabase committed:
Removes caching from lookup-by-id in serialization

Because metadata was missing from the initial declaraion, *lookup-by-id*
was not cached and rebound as intended, and caching this function leads to
errors during import.

john-metabase commented:
Rewriting this branch to just remove the `:dynamic` from `*lookup-by-id*` (now just `lookup-by-id`) since it's inappropriate to cache this function anyway.
